py/mpconfig.h

```diff
@@ -1421,6 +1421,11 @@ typedef double mp_float_t;
 #define MICROPY_PY_SYS_ATEXIT (0)
 #endif
 
+// Whether to allow sys.path to be assigned to (rather than in-place updated)
+#ifndef MICROPY_PY_ALLOW_SYS_PATH_ASSIGNMENT
+#define MICROPY_PY_ALLOW_SYS_PATH_ASSIGNMENT (MICROPY_CONFIG_ROM_LEVEL_AT_LEAST_EXTRA_FEATURES)
```

Comment:
IMO this should always be enabled and not even have a config option.  If a port enables the sys module it gets a mutable sys.path.  Otherwise it's going to get a bit hairy trying to support ports that do not have a mutable version (basically you can't rely on the feature).

Actually, it might be worth making sys.path configurable via `MICROPY_SYS_PATH`.  Then if it is there it must be mutable.  That could be for ports that need sys but for where sys.path doesn't make sense.  But even then, all micropython-lib code should still assume that (a) sys exists, (b) sys.path is mutable.

Reply:
Done - `sys.path` is now configurable via `MICROPY_SYS_PATH` and always assignable.

Also made sys.argv configurable too (even though it's just a regular read-only attribute). (Feels like someone who wants to disable sys.path is probably not interested in sys.argv).